core/txpool/blobpool/blobpool.go

```diff
@@ -873,6 +889,92 @@ func (p *BlobPool) Reset(oldHead, newHead *types.Header) {
 	p.updateStorageMetrics()
 }
 
+// convertSidecars runs on a separate thread and manages the pool lock as needed
+// during the conversion. Any txs that fail to convert are dropped.
+func (p *BlobPool) convertSidecars(txs []*blobTxMeta, cancel chan struct{}) {
+	convert := func(m *blobTxMeta) error {
+		if m.version != types.BlobSidecarVersion0 {
+			return nil
+		}
+		p.lock.RLock()
+		data, err := p.store.Get(m.id)
+		if err != nil {
```

Comment:
you could just unlock rigth after the Get

Reply:
good point